sources/rest_api/config_setup.py

```diff
@@ -163,6 +162,8 @@ def setup_incremental_object(
     for key, value in request_params.items():
         if isinstance(value, dlt.sources.incremental):
             return value, IncrementalParam(start=key, end=None)
+        if isinstance(value, dict) and value.get("type") not in PARAM_TYPES:
```

Comment:
I think we need to find another place for this check: the problem with the check here is that now `setup_incremental_object()` knows about other PARAM_TYPES: which violates encapsulation. A better way is to have the check up the calling stack before setup_incremental_object() gets called.